flow/raster_cache.h

```diff
@@ -296,41 +223,43 @@ class RasterCache {
    * If the number is one, then it must be prepared and drawn on 1 frame
    * and it will then be cached on the next frame if it is prepared.
    */
-  int access_threshold() const { return access_threshold_; }
+  size_t access_threshold() const { return access_threshold_; }
+
+  bool GenerateNewCacheInThisFrame() const {
+    // Disabling caching when access_threshold is zero is historic behavior.
+    return access_threshold_ != 0 &&
+           picture_cached_this_frame_ + display_list_cached_this_frame_ <
+               picture_and_display_list_cache_limit_per_frame_;
+  }
+
+  /**
+   * @brief if the item has been cached return true.
+   * We mean the item will show in the screen, and will create a cache entry if
+   * the item doesn't create cache.
+   * @param id item id
+   * @param matrix
+   */
+  bool MarkSeen(const RasterCacheKeyID& id, const SkMatrix& matrix) const;
+
+  bool UpdateCacheEntry(
+      const RasterCacheKeyID& id,
+      const Context& raster_cache_context,
+      const std::function<void(SkCanvas*)>& render_function) const;
 
  private:
   struct Entry {
     bool used_this_frame = false;
     size_t access_count = 0;
+    // if this entry's parent has cached, we will remove this entry after the
```

Comment:
We probably shouldn't introduce this concept here.

The concept of a survival count has been proposed, but I think the last PR that tried to introduce it is stalled on lack of evidence of its value compared to other fixes.

See https://github.com/flutter/engine/pull/31925

Let's leave survival for another PR.

Reply:
Note that parents caching is not likely to be a short-lived state and waiting 3 frames to make sure they are going to keep caching is probably not going to save many child entries in the long term.

Reply:
It's a concept to keep in mind, but I think it is early to introduce it with this refactoring which is more about consolidating the cache management into a post-Preroll, pre-Paint cycle.